lib/libc/string/memccpy.c

```diff
@@ -38,13 +38,13 @@ __FBSDID("$FreeBSD$");
 #include <string.h>
 
 void *
-memccpy(void *t, const void *f, int c, size_t n)
+memccpy(void * __restrict t, const void * __restrict f, int c, size_t n)
 {
 
 	if (n) {
 		unsigned char *tp = t;
 		const unsigned char *fp = f;
-		unsigned char uc = c;
+		const unsigned char uc = (unsigned char)c;
```

Comment:
This is arguably a separate commit, but just amending the commit message to say you're also explicitly casting 'c' to unsigned char while you are here will suffice.